### PR TITLE
Fix BIP-32 path validation

### PR DESCRIPTION
**Pull request by hmalik88**
Closes https://github.com/MetaMask/MetaMask-planning/issues/713

This PR adds in some extra validation up front to ensure paths are within the `BIP-32` range (`[0, 2 ** 31 - 1]`) per suggestion from Least Authority.

### CODE REVIEW

File: packages/snaps-utils/src/manifest/validation.ts
Comment (FrederikBolding) [inline packages/snaps-utils/src/manifest/validation.ts:52]: Please use `isValidBIP32PathSegment` from `key-tree@7.1.0` (https://github.com/MetaMask/key-tree/blob/main/src/utils.ts#L182).

Currently waiting for approval to be published.

**Review by FrederikBolding — DISMISSED** [review]

File: packages/snaps-utils/src/manifest/validation.ts
Comment (hmalik88) [reply packages/snaps-utils/src/manifest/validation.ts:52]: Might as well takeover the Pr lol

Comment (FrederikBolding) [reply packages/snaps-utils/src/manifest/validation.ts:52]: Sure 😅

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**
